esmvaltool/diag_scripts/galytska23/select_variables_for_tigramite.py

```diff
@@ -0,0 +1,176 @@
+            return psl_sib
+        case 'pressure_aleut':
+            psl_aleut = calculate_slp(item)
+            psl_aleut.var_name = 'Psl_Aleut'
+            return psl_aleut
+        case 'bk_ice':
+            sic_bk = finalize_bk_ice(item)
+            return sic_bk
+        case 'ok_ice':
+            sic_ok = finalize_ok_ice(item)
+            return sic_ok
+        case 'heat_flux':
+            var = prepare_heat_flux(item)
+            return var
+
+
+def calculate_variables(input_dict):
+    """Calculate all necessary variables."""
+    dictionary = {}
+    for key, value in input_dict.items():
+        dictionary.setdefault(key, {})
+        tmp_list = []
+        for item in value:
+            if item['preprocessor'] == "heat_flux":
+                tmp_list.append(variable_cases(item['preprocessor'], item))
+            else:
+                dictionary[key].setdefault(
+                    variable_cases(item['preprocessor'], item).var_name,
+                    variable_cases(item['preprocessor'], item)
+                )
+
+        if key != "HadISST":
+            # calculate heat flux for all data sources except HadISST
+            heat_flux = calculate_heat_flux(tmp_list)
+            dictionary[key].setdefault(heat_flux.var_name, heat_flux)
+    return dictionary
+
+# def plot_selected_timeseries(dictionary):
+#     """Plot timeseries of indicated variables."""
+#     var_names = ['heat_flux', 'PV']
+#     for var in var_names:
+#         plt.figure(figsize=(14, 4))
+#         sns.set_theme()
+#         for key in dictionary:
+#             time_orig = dictionary[key][var].coord('time')
+#             times = np.asarray(time_orig.units.num2date(time_orig.points))
+#             time_pts = [t.strftime('%Y-%m') for t in times]
+#             plt.plot(time_pts, dictionary[key][var].data)
+#             plt.title(var)
+#             plt.ylabel('Anomalies')
+#             plt.xticks(rotation=45, ha="right", rotation_mode='anchor')
+
+
+def run_my_diagnostic(cfg):
+    """Calculate and save final variables into .nc files."""
+    my_files_dict = group_metadata(cfg['input_data'].values(), 'dataset')
+    all_variables = calculate_variables(my_files_dict)
+    for key in my_files_dict:
+        diagnostic_file = get_diagnostic_filename(key, cfg)
+        var = all_variables[key]
+        if key == "ERA5":
+            cube_list = iris.cube.CubeList([
+                var['PV'], var['Arctic_temperature'], var['Psl_Ural'],
+                var['Psl_Sib'], var['Psl_Aleut'], var['heat_flux']])
+        elif key == "HadISST":
+            cube_list = iris.cube.CubeList([
+                var['BK_sic'], var['Ok_sic']])
+        else:
+            cube_list = iris.cube.CubeList([
+                var['PV'], var['Arctic_temperature'], var['Psl_Ural'],
+                var['Psl_Sib'], var['Psl_Aleut'], var['heat_flux'],
+                var['BK_sic'], var['Ok_sic']])
+        iris.save(cube_list, diagnostic_file)
```

Comment:
It would be great to [record the provenance](https://docs.esmvaltool.org/en/latest/community/diagnostic.html#recording-provenance-in-a-python-diagnostic-script) for these files. Let me know if I can help you with that.